sklearn/exceptions.py

```diff
@@ -0,0 +1,144 @@
+    score on this train-test partition for these parameters will be set to
+    0.000000. Details: \\\\nValueError(\\'Penalty term must be positive; got
+    (C=-2)\\',)",)'
+    """
+
+
+class ChangedBehaviorWarning(UserWarning):
+    """Warning class used to notify the user of any change in the behavior.
+
+    Examples
+    --------
+    >>> from sklearn.grid_search import GridSearchCV
+    >>> from sklearn.svm import LinearSVC
+    >>> from sklearn.metrics import f1_score
+    >>> from sklearn.datasets import make_blobs
+    >>> from sklearn.exceptions import ChangedBehaviorWarning
+    >>> import warnings
+    >>> warnings.simplefilter('always', ChangedBehaviorWarning)
+    >>> gs = GridSearchCV(estimator=LinearSVC(random_state=0),
+    ...                   param_grid={'C': [1, 10, 100]}, scoring='f1_micro')
+    >>> X, y = make_blobs(random_state=0)
+    >>> with warnings.catch_warnings(record=True) as w:
+    ...     score = gs.fit(X[:20], y[:20]).score(X[20:], y[20:])
+    ...     print("The warning raised was : %r" % repr(w[-1].message))
+    ...     print("The score is : %s" % score)
+    ...                        # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+    The warning raised was : 'ChangedBehaviorWarning("The long-standing
+    behavior to use the estimator\\'s score function in GridSearchCV.score
+    has changed. The scoring parameter is now used.",)'
+    The score is : 0.8625
+    """
+
+
+class DataConversionWarning(UserWarning):
+    """Warning used to notify implicit data conversions happening in the code.
+
+    This warning is raised when the data, usually the input, is implicitly
+    converted/copied based on the requirements.
+
+    Examples
+    --------
+    >>> from sklearn.exceptions import DataConversionWarning
+    >>> from sklearn.preprocessing import scale
+    >>> from sklearn.exceptions import DataConversionWarning
+    >>> import numpy as np
+    >>> import warnings
+    >>> warnings.simplefilter('always', DataConversionWarning)
+    >>> X = np.array([[1, 2, 0],
+    ...               [0, 0, 0]], dtype=np.uint8)
+    >>> with warnings.catch_warnings(record=True) as w:
+    ...     scaled_X = scale(X)
+    ...     print("The warning raised was : %r" % repr(w[-1].message))
+    ...     print("The scaled X is : ")
+    ...     print(scaled_X)
+    ...                        # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+    The warning raised was : "DataConversionWarning('Data with input dtype
+    uint8 was converted to float64 by the scale function',)"
+    The scaled X is :
+    [[ 1.  1.  0.]
+     [-1. -1.  0.]]
+    """
+
+
+class EfficiencyWarning(UserWarning):
+    """Warning used to notify the user that the efficiency is affected.
+
+    This warning notifies the user that the efficiency may not be optimal due
+    to some reason which may be included as a part of the warning message.
+
+    Examples
+    --------
+    >>> from sklearn.utils.extmath import _fast_dot
```

Comment:
I don't think doctesting is really relevant here. The class itself is not functional in this example.